test/e2e/rc.go

```diff
@@ -235,3 +252,91 @@ func rcConditionCheck(f *framework.Framework) {
 	}
 	Expect(err).NotTo(HaveOccurred())
 }
+
+func testRCAdoptMatchingOrphans(f *framework.Framework) {
+	name := "pod-adoption"
+	By(fmt.Sprintf("Given a Pod with a 'name' label %s is created", name))
+	p := f.PodClient().CreateSync(&v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"name": name,
+			},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  name,
+					Image: nginxImageName,
+				},
+			},
+		},
+	})
+
+	By("When a replication controller with a matching selector is created")
+	replicas := int32(2)
+	rcSt := newRCWithSelector(name, replicas, map[string]string{"name": name}, name, nginxImageName, name)
+	rc, err := f.ClientSet.Core().ReplicationControllers(f.Namespace.Name).Create(rcSt)
+	Expect(err).NotTo(HaveOccurred())
+	// Cleanup the ReplicationController when we are done.
+	defer func() {
+		if err := framework.DeleteRCAndPods(f.ClientSet, f.InternalClientset, f.Namespace.Name, rc.Name); err != nil {
+			framework.Logf("Failed to cleanup ReplicationController %v: %v.", rc.Name, err)
+		}
+	}()
+
+	By("Then the orphan pod is adopted")
+	err = wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		p2, err := f.ClientSet.Core().Pods(f.Namespace.Name).Get(p.Name, metav1.GetOptions{})
```

Comment:
Can you just do `Get(p.Name, nil)`? Though i think either way it's enough to bypass the cache.

My first comment regarding checking IsNotFound is still valid.

@kargakis want to make sure we are talking about the same cache here. I'm thinking about this [one](https://github.com/kubernetes/kubernetes/blob/v1.6.0-beta.4/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L312).